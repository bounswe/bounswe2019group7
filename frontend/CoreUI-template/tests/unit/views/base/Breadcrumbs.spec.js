import Vue from 'vue'
import { shallowMount, mount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Breadcrumbs from '@/views/base/Breadcrumbs'

Vue.use(BootstrapVue)

describe('Breadcrumbs.vue', () => {
  it('has a name', () => {
    expect(Breadcrumbs.name).toMatch('breadcrumbs')
  })
  it('has a created hook', () => {
    expect(typeof Breadcrumbs.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof Breadcrumbs.data).toMatch('function')
    const defaultData = Breadcrumbs.data()
    expect(typeof defaultData.items).toMatch('object')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Breadcrumbs)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Breadcrumbs', () => {
    const wrapper = shallowMount(Breadcrumbs)
    expect(wrapper.is(Breadcrumbs)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(Breadcrumbs)
    expect(wrapper.find('header.card-header > div > strong').text()).toMatch('Bootstrap Breadcrumb')
  })
  test('renders correctly', () => {
    const wrapper = mount(Breadcrumbs)
    expect(wrapper.element).toMatchSnapshot()
  })
})
