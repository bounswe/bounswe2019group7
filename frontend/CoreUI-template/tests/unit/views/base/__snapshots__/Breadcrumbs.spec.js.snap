// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs.vue renders correctly 1`] = `
<div
  class="wrapper"
>
  <div
    class="animated fadeIn"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="card"
        >
          <!---->
          <header
            class="card-header"
          >
            <div>
              <i
                class="fa fa-align-justify"
              />
              <strong>
                 Bootstrap Breadcrumb
              </strong>
               
              <div
                class="card-header-actions"
              >
                <a
                  class="card-header-action"
                  href="https://bootstrap-vue.js.org/docs/components/breadcrumb"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  <small
                    class="text-muted"
                  >
                    docs
                  </small>
                </a>
              </div>
            </div>
          </header>
          <div
            class="card-body"
          >
            <!---->
            <!---->
             
            <ol
              class="breadcrumb"
            >
              <li
                class="breadcrumb-item"
              >
                <a
                  class=""
                  href="#"
                  target="_self"
                >
                  Admin
                </a>
              </li>
              <li
                class="breadcrumb-item"
              >
                <a
                  class=""
                  href="#"
                  target="_self"
                >
                  Manage
                </a>
              </li>
              <li
                class="breadcrumb-item active"
              >
                <span
                  aria-current="location"
                >
                  Library
                </span>
              </li>
            </ol>
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;
